unison-cli/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -2311,7 +2319,7 @@ projectClone =
     { patternName = "project.clone",
       aliases = [],
       visibility = I.Hidden,
-      argTypes = [(Required, projectAndBranchNamesArg)],
+      argTypes = [],
```

Comment:
Don't we want an element for these? `noCompletionsArg` at least, or is it a fzf thing to leave it off?

Reply:
It turns out (I'm 95% sure) these things are only used for tab-completion. I went through and removed some in this PR because I discovered them to be inaccurate. This one, for example, shouldn't take the same `projectAndBranchNamesArg` as `switch` does because we don't want to populate `project.clone` tab completion with local projects and branches.

So, in summary, for commands that don't have any tab-completion yet, I think `argTypes` can be anything. I didn't know about `noCompletionsArg` when I set it to the empty list.

But do you happen to know if something like `[(Optional, noCompletionsArg)]` does anything different than `[]`? As far as I can tell there's no relevance to that `Optional`, `Required`, `ZeroPlus`, etc... all of that is just controlled by the actual parser. I could definitely have missed something, though.

Reply:
I kind of forget but I think there was (at some point) an element of counting args or knowing how many args a command took, but I could believe it's now vestigial. It may always have been unused.

Reply:
> we don't want to populate project.clone tab completion with local projects and branches.

We want to populate it with remote projects and branches right? (though we can't currently support that right? or can we already do IO when generating completions?)

Reply:
Yeah, that sounds right. And we can do database and http requests in the completer.